Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp

```diff
@@ -319,17 +319,17 @@ void webKitMediaSrcEmitStreams(WebKitMediaSrc* source, const Vector<RefPtr<Media
     gst_element_post_message(GST_ELEMENT(source), gst_message_new_stream_collection(GST_OBJECT(source), source->priv->collection.get()));
 
     for (const RefPtr<Stream>& stream: source->priv->streams.values()) {
-        // Workaround: gst_element_add_pad() should already call gst_pad_set_active() if the element is PAUSED or
-        // PLAYING. Unfortunately, as of GStreamer 1.18.2 it does so with the element lock taken, causing a deadlock
-        // in gst_pad_start_task(), who tries to post a `stream-status` message in the element, which also requires
-        // the element lock. Activating the pad beforehand avoids that codepath.
-        // https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/210
-        // FIXME: Remove this workaround when the bug gets fixed and versions without the bug are no longer in use.
-        GstState state;
-        gst_element_get_state(GST_ELEMENT(source), &state, nullptr, 0);
-        if (state > GST_STATE_READY)
-            gst_pad_set_active(GST_PAD(stream->pad.get()), true);
-
+        if (!webkitGstCheckVersion(1, 20, 6)) {
+            // Workaround: gst_element_add_pad() should already call gst_pad_set_active() if the element is PAUSED or
+            // PLAYING. Unfortunately, as of GStreamer 1.18.2 it does so with the element lock taken, causing a deadlock
```

Comment:
> as of GStreamer 1.18.2

Now that the fix is released the comment should be updated to "versions of GStreamer prior to 1.20.6 (stable) / 1.21.x (unstable)".

Reply:
> This check would disable the workaround in versions of 1.21 prior to the fix, which are still affected by the bug.

1.21 is a development version, distros are not shipping development versions, only stable releases.

Reply:
People still install development versions from time to time though.

Reply:
Well. At some point we need to find a balance between keeping technical debt at bay and pleasing every wild setup across the globe...

> People still install development versions from time to time though.

Of an old development branch? I don't think we should support that.
We usually do development version checks for the current development branch, not for versions that have been shipped as stable.

Reply:
1.20-1.21 is not that old. And in this case the consequence of a false positive is a deterministic deadlock.

Reply:
What's wrong with the current implementation?

Reply:
Sorry but I don't see the need to over-engineer this function. People installing unstable old GStreamer versions should not expect stable runtime behavior...

Reply:
Fwiw, I think we should not try to support unreleased versions fmpov.

Reply:
>  Fwiw, I think we should not try to support unreleased versions fmpov.

Yet we have version checks for unstable versions in several parts of the code :confused: 

```
$ ggrep 'webkitGstCheckVersion(1, [0-9]*[13579],'
WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp
bool GStreamerMediaEndpoint::initializePipeline()
    if (webkitGstCheckVersion(1, 21, 0)) {
UniqueRef<GStreamerDataChannelHandler> GStreamerMediaEndpoint::findOrCreateIncomingChannelHandler(GRefPtr<GstWebRTCDataChannel>&& dataChannel)
    if (!webkitGstCheckVersion(1, 21, 0))

WebCore/Modules/mediastream/gstreamer/GStreamerStatsCollector.cpp
static gboolean fillReportCallback(GQuark, const GValue* value, gpointer userData)
        if (webkitGstCheckVersion(1, 21, 0)) {
        if (webkitGstCheckVersion(1, 21, 0)) {

WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp
void registerWebKitGStreamerElements()
        if (webkitGstCheckVersion(1, 21, 0)) {
```

Reply:
Looks like your checkout is outdated, the only file that needs updating is Source/WebCore/Modules/mediastream/gstreamer/GStreamerStatsCollector.cpp

Reply:
> Fwiw, I think we should not try to support unreleased versions fmpov.

I don't think that's avoidable when we want to use functionality not shipped in a stable gst release yet though.

Reply:
I'll accept the patch...